src/components/ElTelInput.vue

```diff
@@ -128,6 +128,17 @@ export default {
       return this.$attrs.size || 'mini';
     }
   },
+  watch: {
+    value: {
+      deep: true,
+      handler(newValue) {
+        const parsedPhoneNumber = getParsedPhoneNumber(newValue.nationalNumber, newValue.country);
+        this.countryCallingCode = parsedPhoneNumber.countryCallingCode || this.countryCallingCode;
+        this.country = parsedPhoneNumber.country || this.defaultCountry || this.country;
```

Comment:
this should probably use the existing value before the default country
```suggestion
        this.country = parsedPhoneNumber.country || this.country || this.defaultCountry;
```